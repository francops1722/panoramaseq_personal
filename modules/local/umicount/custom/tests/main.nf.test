nextflow_process {

    name "Test Process UMICOUNT"
    script "../main.nf"
    process "UMICOUNT"
    config "./nextflow.config"
    tag "modules"
    tag "modules_local"
    tag "umi_count"

    test("panoramaseq [bam] - S09") {

        when {
            params {
                panoramaseq_testdata_base_path = "https://raw.githubusercontent.com/nf-core/test-datasets/panoramaseq/testdata/"
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'S09', N_barcodes:34500, sample_size:10000, len_barcode:36, Nthresh:9, Ntriage:1000 ], // meta map
                    file(params.panoramaseq_testdata_base_path + 'samtools/test1.assigned_sorted.bam', checkIfExists: true),
                    file(params.panoramaseq_testdata_base_path + 'samtools/test1.assigned_sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.umi_counts != null },
                { assert process.out.log_files != null },
                { assert process.out.versions != null },
                { assert snapshot(
                    process.out.umi_counts,
                    process.out.versions
                ).match() }
            )
        }

    }

    test("panoramaseq [bam] - S09 - stub") {

        options "-stub"

        when {
            params {
                panoramaseq_testdata_base_path = "https://raw.githubusercontent.com/nf-core/test-datasets/panoramaseq/testdata/"
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'S09', N_barcodes:34500, sample_size:10000, len_barcode:36, Nthresh:9, Ntriage:1000 ], // meta map
                    file(params.panoramaseq_testdata_base_path + 'samtools/test1.assigned_sorted.bam', checkIfExists: true),
                    file(params.panoramaseq_testdata_base_path + 'samtools/test1.assigned_sorted.bam.bai', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert process.out.umi_counts != null },
                { assert process.out.log_files != null },
                { assert process.out.versions != null },
                { assert snapshot(
                    process.out.umi_counts,
                    process.out.versions
                ).match("stub") }
            )
        }

    }

}
