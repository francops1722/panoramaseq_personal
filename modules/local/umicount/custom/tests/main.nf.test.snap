{
    "panoramaseq [bam] - S09": {
        "content": [
            [
                [
                    {
                        "id": "S09",
                        "N_barcodes": 34500,
                        "sample_size": 10000,
                        "len_barcode": 36,
                        "Nthresh": 9,
                        "Ntriage": 1000
                    },
                    "S09_counts.tsv.gz:md5,cf4ef97b98ddf460d9d159364afde8f4"
                ]
            ],
            [
                "versions.yml:md5,95af6601997fd9222f34687cfe3ba1dc"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-08-12T18:05:25.19933596"
    },
    "stub": {
        "content": [
            [
                [
                    {
                        "id": "S09",
                        "N_barcodes": 34500,
                        "sample_size": 10000,
                        "len_barcode": 36,
                        "Nthresh": 9,
                        "Ntriage": 1000
                    },
                    "S09_counts.tsv.gz:md5,965f2e431a994b442413018f09f0a3d9"
                ]
            ],
            [
                "versions.yml:md5,22c255898d2bad457e64434971232ba5"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-08-12T18:05:29.94539206"
    }
}